tools/include/eosio/gen.hpp

```diff
@@ -615,5 +627,25 @@ struct generation_utils {
       if (get_base_type_name(t).find("<") != std::string::npos) return false;
       return get_base_type_name(t).compare(get_type_alias_string(t)) != 0;
    }
+
+   inline bool is_eosio_non_unique(const clang::QualType& t) {
+      auto str_name = t.getAsString();
+      auto nu_re = std::regex("eosio::non_unique<[a-zA-Z]+[a-zA-Z0-9].*>");
```

Comment:
We don't really care what the template name is as this simply a predicate for the template name.  There's no need to build up the regex for these. 

```c++
constexpr std::string_view test_string = "eosio::non_unique<";
return t.getAsString().substr(0, test_string.size()) == test_string;
```